src/target/riscv/riscv-013.c

```diff
@@ -1911,6 +1927,135 @@ static int set_group(struct target *target, bool *supported, unsigned int group,
 	return ERROR_OK;
 }
 
+static int wait_for_idle_if_needed(struct target *target)
+{
+	dm013_info_t *dm = get_dm(target);
+	if (!dm)
+		return ERROR_FAIL;
+	if (!dm->abtractcs_busy)
+		return ERROR_OK;
+
+	uint32_t abstractcs;
+	int result = wait_for_idle(target, &abstractcs);
```

Comment:
Putting comments here may help the reader understand the broader context of this function:

```suggestion
	if (!dm->abtractcs_busy)
		/* The previous abstract command ended correctly
		 * and busy was cleared. No need to do anything. */
		return ERROR_OK;

	/* The previous abstract command timed out and abstractcs.busy 
	 * may have remained set. Wait for it to get cleared. */
	uint32_t abstractcs;
	int result = wait_for_idle(target, &abstractcs);
```

Reply:
Great clarification! Addressed.